ivy_tests/test_ivy/test_functional/test_core/test_creation.py

```diff
@@ -149,6 +149,49 @@ def test_logspace(start_n_stop_n_num_n_base_n_axis, dtype, tensor_fn, device, ca
 # empty()
 # empty_like()
 # eye()
+
+@given(
+    n_rows=st.integers(min_value=0, max_value=5),
+    n_cols=st.none() | st.integers(min_value=0, max_value=5),
+    k=st.integers(min_value=-5, max_value=5),
+    batch_shape=helpers.lists(st.integers(1, 5), min_size=1, max_size=1),
```

Comment:
we should diversify the list length more, this is always a list of length 1.

Reply:
@djl11 can you please look into `torch` and `tensorflow` backend implementation of `eye()`. I am a bit confused over `torch` but `tensorflow` is acting weird as well.